Remove deprecated gdk_screen_get_width/height

dylanmccall opened the pull request:
To achieve this, eos_is_composite_tv_screen is replaced with a
GdkMonitor equivalent: eos_is_composite_tv_monitor.

With this change, it is possible for EosWindow to better detect the
physical monitor it is displayed on and adapt accordingly. However,
the existing code in eoswindow.c does not make use of this
functionality.

https://phabricator.endlessm.com/T28284

dylanmccall committed:
Remove deprecated gdk_screen_get_width/height

To achieve this, eos_is_composite_tv_screen is replaced with a
GdkMonitor equivalent: eos_is_composite_tv_monitor.

With this change, it is possible for EosWindow to better detect the
physical monitor it is displayed on and adapt accordingly. However,
the existing code in eoswindow.c does not make use of this
functionality.

https://phabricator.endlessm.com/T28284

wjt commented:
I suggest dropping this function instead – we dropped composite mode support in the OS. https://phabricator.endlessm.com/T22102

dylanmccall committed:
Completely remove eos_is_composite_tv_monitor

Turns out composite mode support has been removed in EOS, so no sense
doing it here.

https://phabricator.endlessm.com/T28284

dylanmccall commented:
Ack! That's good to know :) Tweaked accordingly. We do apply a few composite-mode-specific styles in eos-knowledge-lib, so there's some dead code to remove there as well, but nothing huge.

wjt commented on endless/endless.h:
```diff
@@ -26,7 +26,7 @@ EOS_SDK_DEPRECATED_IN_0_0
                                     GError **error);
 
-EOS_SDK_AVAILABLE_IN_0_6
+EOS_SDK_DEPRECATED_IN_0_6
```
Should we consider removing this function entirely, or just making it `return FALSE`?

dylanmccall replied:
I'd rather keep it to avoid mishaps for now, but that's a good thought about returning FALSE. I added that and squashed it into the previous commit.
